internal/crypto/internal_ca.go

```diff
@@ -108,112 +105,6 @@ func makeSelfSignedCAConfig(subject pkix.Name, caLifetime time.Duration) (*oscry
 	return caConfig, nil
 }
 
-type CertificateExtensionFunc func(*x509.Certificate) error
-
-func (ca *CA) MakeServerCert(hostnames []string, expiryDays int, fns ...CertificateExtensionFunc) (*TLSCertificateConfig, error) {
-	if len(hostnames) < 1 {
-		return nil, fmt.Errorf("at least one hostname must be provided")
-	}
-
-	serverPublicKey, serverPrivateKey, publicKeyHash, _ := NewKeyPairWithHash()
-
-	now := time.Now()
-	serverTemplate := &x509.Certificate{
-		Subject: pkix.Name{CommonName: hostnames[0]},
-
-		SignatureAlgorithm: x509.ECDSAWithSHA256,
-
-		NotBefore:    now.Add(-1 * time.Second),
-		NotAfter:     now.Add(time.Duration(expiryDays) * 24 * time.Hour),
-		SerialNumber: big.NewInt(1),
-
-		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
-		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
-		BasicConstraintsValid: true,
-
-		AuthorityKeyId: ca.Config.Certs[0].SubjectKeyId,
-		SubjectKeyId:   publicKeyHash,
-	}
-	serverTemplate.IPAddresses, serverTemplate.DNSNames = oscrypto.IPAddressesDNSNames(hostnames)
-	for _, fn := range fns {
-		if err := fn(serverTemplate); err != nil {
-			return nil, err
-		}
-	}
-
-	serverCrt, err := ca.signCertificate(serverTemplate, serverPublicKey)
-	if err != nil {
-		return nil, err
-	}
-	server := &TLSCertificateConfig{
-		Certs: append([]*x509.Certificate{serverCrt}, ca.Config.Certs...),
-		Key:   serverPrivateKey,
-	}
-	return server, nil
-}
-
-func (ca *CA) MakeClientCertificate(certFile, keyFile string, subject string, expiryDays int) (*TLSCertificateConfig, error) {
```

Comment:
Same with this one

Reply:
Yes. 

Phase 1 - move all private key operations to internal_ca.go

We get an ugly, but functional "plugin" - all ops in one place

Phase 2 - remove all private key ops in signing. Some wrappers pop up from there and end up in the outer. 

Phase 3 - API is cleaned up for pluggability